frontend/components/SetupPage/FundYourAgent/hooks/useGetRefillRequirementsWithMonthlyGas.ts

```diff
@@ -0,0 +1,198 @@
+import { useEffect, useMemo } from 'react';
+
+import { AddressBalanceRecord, MasterSafeBalanceRecord } from '@/client';
+import { getTokenDetails } from '@/components/Bridge/utils';
+import {
+  ChainTokenConfig,
+  getNativeTokenSymbol,
+  TOKEN_CONFIG,
+  TokenConfig,
+} from '@/config/tokens';
+import { AddressZero } from '@/constants/address';
+import { EvmChainId } from '@/constants/chains';
+import { SERVICE_TEMPLATES } from '@/constants/serviceTemplates';
+import { TokenSymbolConfigMap, TokenSymbolMap } from '@/constants/token';
+import { useBalanceAndRefillRequirementsContext } from '@/hooks/useBalanceAndRefillRequirementsContext';
+import { useMasterWalletContext } from '@/hooks/useWallet';
+import { Address } from '@/types/Address';
+import { AgentConfig } from '@/types/Agent';
+import { bigintMax } from '@/utils/calculations';
+import { formatUnitsToNumber } from '@/utils/numberFormatters';
+
+import { useBeforeBridgeFunds } from '../../Create/SetupEoaFunding/useBeforeBridgeFunds';
+import { TokenRequirement } from '../TokensRequirements';
+
+const ICON_OVERRIDES: Record<string, string> = {
+  [TokenSymbolMap['XDAI']]: '/tokens/wxdai-icon.png',
+};
+
+const getTokenMeta = (
+  tokenAddress: Address,
+  chainConfig: ChainTokenConfig,
+  evmHomeChainId: EvmChainId,
+) => {
+  if (tokenAddress === AddressZero) {
+    const symbol = getNativeTokenSymbol(evmHomeChainId);
+    const config = chainConfig[symbol];
+    return {
+      symbol,
+      decimals: config.decimals,
+      iconSrc: TokenSymbolConfigMap[symbol].image || '',
+    };
+  }
+
+  const tokenDetails = getTokenDetails(
```

Comment:
I think getTokenDetails also checks for native token, can this be used directly without the `if (tokenAddress === AddressZero) ` part?

Reply:
<img width="787" height="247" alt="Screenshot 2025-08-29 at 5 08 52 PM" src="https://github.com/user-attachments/assets/2d2364ab-ff6b-47aa-a8b8-79e4525df1b8" />
ah unfortunately it won't handle for XDAI, i'm not sure if should update the function in order to handle it